code/Company.php

```diff
@@ -100,15 +102,13 @@ public function getCMSFields()
         return $fields;
     }
 
-    public function validate()
+    public function getCMSCompositeValidator(): CompositeValidator
     {
-        $result = parent::validate();
-
-        if (!$this->Name) {
-            $result->addFieldError('Name', 'Name is required');
-        }
-
-        return $result;
+        $validator = parent::getCMSCompositeValidator();
+        $validator->addValidator(RequiredFields::create([
+            'Name',
+        ]));
+        return $validator;
     }
```

Comment:
This might affect existing behat tests somewhere - I wouldn't be comfortable merging this without seeing a full kitchen sink CI with all behat tests being run

Reply:
Ok. I'll run tests on kitchen-sink.

Reply:
You'll need to find out why those behat runs are failing - the fact that they're not running _at all_ means I can't be sure that they'll pass with this change.

Reply:
I have fixed all related failed tests.
- https://github.com/sabina-talipova/recipe-kitchen-sink/actions/runs/6805275693/
But there are anyway another failed test that are not related to this PR.